src/Process/Builder/MutantProcessBuilder.php

```diff
@@ -75,6 +87,21 @@ public function createProcessForMutant(Mutant $mutant, string $testFrameworkExtr
             $process->inheritEnvironmentVariables();
         }
 
-        return new MutantProcess($process, $mutant);
+        $mutantProcess = new MutantProcess($process, $mutant);
+
+        $eventDispatcher = $this->eventDispatcher;
+        $resultFactory = $this->resultFactory;
+
+        $mutantProcess->registerTerminateProcessClosure(static function () use (
+            $mutantProcess,
+            $eventDispatcher,
+            $resultFactory
+        ): void {
+            $eventDispatcher->dispatch(new MutantProcessWasFinished(
+                $resultFactory->createFromProcess($mutantProcess))
+            );
+        });
```

Comment:
Do you think MutantProcess can't be aware of the events? E.g. it can do what it should without closure DI.

I'm not sure this is the right way, but adding a callback with an event here, which events are used to avoid these callbacks, looks a tiny bit messy.

Reply:
I mean, would it be simpler to inject `eventDispatcher` and `resultFactory` into the `MutantProcess`, rather then pulling these tricks with closures and such?

Reply:
Not to say we should go the way right this moment. For one because I come to thinking it'll be better to inject some sort of a result handler, which, in turn, would dispatch the events.

Reply:
My reasoning was that:

- The current dependency does not add up: the runner needs to be aware of the underlying `ProcessBearer` implementation - its callback registered only works with one
- It is common for a process to have a callback (in fact the Symfony one has one, it's just awkward to try to extend it)

But I would be lying to say I'm 100% confident on that one... I just think it's slightly more correct, without feeling that I nailed it either